rust/src/scanner/scan_runner.rs

```diff
@@ -318,49 +316,41 @@ exit({rc});
             ("description".to_owned(), true.into()),
             ("OPENVAS_VERSION".to_owned(), "testus".into()),
         ];
-        let storage = DefaultDispatcher::new();
+        let storage = Arc::new(InMemoryStorage::new());
 
         let register = Register::root_initial(&initial);
         let target = ContextTarget::default();
         let functions = nasl_std_functions();
         let loader = |_: &str| code.to_string();
-        let key = ContextKey::FileName(id.to_string());
+        let key = ScanID(id.to_string());
```

Comment:
How do you do differentiation between a normal scan (actual scan id) and parsing metadata (filename that is also a value) when you handle it as a scan_id?

Reply:
The filename information is now set by the context when storing a VT. Two lines below, the filename is now given to the Context as a string. So nothing changed here.